common/dynamicconfig/basic_client.go

```diff
@@ -198,8 +181,15 @@ func (bc *basicClient) getValueWithFilters(
 	defaultValue interface{},
 ) (interface{}, error) {
 	keyName := strings.ToLower(key.String())
-	values := bc.values.Load().(configValueMap)
+	values := bc.getValues()
 	constrainedValues := values[keyName]
+
+	if defaultConstraints, ok := defaultValue.([]ConstrainedValue); ok {
+		// if defaultValue is a list of constrained values, then one of them must have an empty
+		// constraint set
+		constrainedValues = append(slices.Clone(constrainedValues), defaultConstraints...)
```

Comment:
Is this allocated every time GetTaskQueueRead/WritePartitionCount is called?

Reply:
yeah :(  but keep in mind, those already allocate a slice of four maps(!). I have a plan to turn those all into structs and do some more allocation profiling, because it's a mess right now. but for now it seemed okay.

an alternative here is to duplicate the code instead: loop over defaultConstraints after each loop over constrainedValues. I'm open to that. though in that case I would also want to make one more change: require callers to always pass in the empty filter at the end of `filters`, so that we can get rid of the loop in line 203-208.

Reply:
I just realized that getValueWithFilters is called for all dynamic config keys, so it is called very often, many per each request.